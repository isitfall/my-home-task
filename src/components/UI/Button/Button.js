import React from 'react';
import PropTypes from 'prop-types';

export default function Button(props) {
    return (
        <button
            type={props.buttonType}
            className={props.styles}
            onClick={props.click}>
            {props.title}
        </button>
    )
}

Button.propTypes = {
    styles: PropTypes.string,
    title: PropTypes.string,
    click: PropTypes.func,
    buttonType: PropTypes.string
}