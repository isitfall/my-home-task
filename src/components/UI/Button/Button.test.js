import React from 'react';
import { screen, render, } from "@testing-library/react";
import Button from "./Button";

it('should render the Button', function () {
    const { asFragment } = render(<Button title={'click me'} className={'btn-primary'} type={'submit'}/>);

    expect(asFragment()).toMatchSnapshot();
    screen.debug()
}); 