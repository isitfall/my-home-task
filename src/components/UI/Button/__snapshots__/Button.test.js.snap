// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Button 1`] = `
<DocumentFragment>
  <button>
    click me
  </button>
</DocumentFragment>
`;
